httpserver/bin/basic_writer_server.dart

```diff
@@ -9,36 +9,37 @@
 import 'dart:io';
 import 'dart:convert';
 
-void main() {
+main() async {
+  var server =
+      await HttpServer.bind(InternetAddress.LOOPBACK_IP_V4, 4049);
+  await for (var req in server) {
+    ContentType contentType = req.headers.contentType;
+    BytesBuilder builder = new BytesBuilder();
```

Comment:
I would move this declaration closer to where it's used. Probably just inside the `try` body below.